concordium-consensus/src/Concordium/Scheduler/TreeStateEnvironment.hs

```diff
@@ -475,13 +489,15 @@ executeFrom blockHash slotNumber slotTime blockParent blockBaker mfinInfo newSee
               _maxBlockEnergy = maxBlockEnergy,
               _accountCreationLimit = chainParams ^. cpAccountCreationLimit
               }
-        (res, finState) <- runBSM (Sch.runTransactions txs) context (mkInitialSS bshandle1 :: LogSchedulerState m)
+        startingCache <- getTransactionVerificationCache
+        (res, finState) <- runBSM (Sch.runTransactions txs) context (mkInitialSS bshandle1 startingCache :: LogSchedulerState m)
         let usedEnergy = finState ^. schedulerEnergyUsed
         let bshandle2 = finState ^. schedulerBlockState
+        let finalCache = finState ^. transactionVerificationCache
```

Comment:
Is this even different from the startingCache?

I don't see the scheduler updating it with anything. At some point the idea was that if we discover new information during block execution (such as correct signature with new keys) we'd update it. But this does not seem to be done.

Reply:
Correct.